include/deal.II/non_matching/quadrature_generator.h

```diff
@@ -456,6 +477,85 @@ namespace NonMatching
      * This quadrature always contains zero points in 1D.
      */
     const ImmersedSurfaceQuadrature<0, 1> surface_quadrature;
+
+    /**
+     * Function that describes our level set function in reference space.
+     */
+    std::unique_ptr<
+      internal::QuadratureGeneratorImplementation::CellWiseFunction<1>>
+      reference_space_level_set;
+  };
+
+
+  /**
+   * This class creates immersed quadrature rules of cells in reference space
+   * when the domain is described by a discrete level set vector.
+   */
```

Comment:
I think this documentation could be extended slightly. Here is a suggestion:
```suggestion
 /**
   * This class generates the same type of immersed quadrature rules as those
   * described in the QuadratureGenerator class. The difference is that this
   * class handles the case when the the domain is a discrete level set
   * function, i.e., when the level set function is described as a
   * (DoFHandler, Vector) pair. The generate()-function of this class takes a
   * cell in real space and constructs the immersed quadrature rules in
   * reference space over this cell. These quadrature rules can then be obtained
   * with one of the functions:
   * get_inside_quadrature(),
   * get_outside_quadrature(), and
   * get_surface_quadrature().
   *
   * Internally, the quadrature generation is done by transforming the discrete
   * level set function from real space to reference space and using the same
   * algorithm as in the QuadratureGenerator class.
   */
```